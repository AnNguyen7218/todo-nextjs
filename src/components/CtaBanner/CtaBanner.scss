@import 'config.colors';
@import 'config.gutters';
@import 'config.media';

.wrapper {
  padding-top: gutter(24);
  padding-bottom: gutter(24);
}

.container {
  display: flex;
  flex-direction: column;
  background-color: $black;
  padding: gutter(48) gutter(24) gutter(24) gutter(24);

  @include break(md) {
    flex-direction: row;
    align-items: flex-start;
    padding-top: gutter(0);
    padding-bottom: gutter(0);
  }

  @include break(xl) {
    padding-right: gutter(96);
    padding-left: gutter(96);
  }
}

.title-container {
  flex: 2;
  height: 100%;
  background-color: $black;
  color: $white;
  margin-bottom: gutter(24);

  @include break(md) {
    margin-right: gutter(20);
    margin-bottom: gutter(48);
    margin-top: gutter(48);
  }

  @include break(xl) {
    margin-top: gutter(96);
    margin-bottom: gutter(96);
  }
}

.header {
  margin-bottom: gutter(24);

  @include break(xl) {
    margin-bottom: gutter(48);
  }
}

.image-section {
  display: flex;
  justify-content: flex-start;
  flex: 1;
  width: 100%;
  background-color: $black;

  @include break(md) {
    height: 100%;
    margin-top: gutter(24);
    margin-bottom: gutter(24);
  }

  @include break(xl) {
    margin-top: gutter(40);
    margin-bottom: gutter(40);
  }
}

.img {
  width: 22rem;
  height: 22rem;

  @include break(md) {
    width: 25rem;
    height: 25rem;
  }

  @include break(xl) {
    width: 40rem;
    height: 40rem;
  }
}
