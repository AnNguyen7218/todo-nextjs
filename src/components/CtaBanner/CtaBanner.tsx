import React from 'react';
import image from 'assets/images/cta-banner.png';
import { Container, Button } from 'components';
import css from './CtaBanner.scss';

const CtaBanner = () => {
  return (
    <Container className={css['wrapper']}>
      <div className={css['container']}>
        <div className={css['title-container']}>
          <h2 className={css['header']}>
            Reach a global customer base for your professional services
          </h2>
          <Button buttonType="fixed" theme="containedColor" color="white">
            Get started
          </Button>
        </div>
        <div className={css['image-section']}>
          <img src={image} className={css['img']} />
        </div>
      </div>
    </Container>
  );
};

export default CtaBanner;
