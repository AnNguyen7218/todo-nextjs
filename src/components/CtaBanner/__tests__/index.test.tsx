import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CtaBanner from '../CtaBanner';

afterEach(cleanup);

describe('CtaBanner Component', () => {
  test('CtaBanner should match snapshot', () => {
    const { container } = render(<CtaBanner />);
    expect(container).toMatchSnapshot();
  });
});
