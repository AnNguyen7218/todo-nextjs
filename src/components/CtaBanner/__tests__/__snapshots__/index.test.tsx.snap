// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaBanner Component CtaBanner should match snapshot 1`] = `
<div>
  <div
    class="container wrapper"
  >
    <div
      class="container"
    >
      <div
        class="title-container"
      >
        <h2
          class="header"
        >
          Reach a global customer base for your professional services
        </h2>
        <button
          class="button button--fixed button--containedColor button--white"
          type="button"
        >
          <div
            class="button-children"
          >
            Get started
          </div>
        </button>
      </div>
      <div
        class="image-section"
      >
        <img
          class="img"
          src="cta-banner.png"
        />
      </div>
    </div>
  </div>
</div>
`;
